sources/parsers/src/Evaluator.cpp

```diff
@@ -0,0 +1,202 @@
+/// @file parsers/src/Evaluator.cpp
+///
+/// The class provides means for evaluating user format expressions.
+///
+/// @par Full Description
+/// The class provides means for evaluating user format expressions.
+/// It is a facade to Flex/Bison generated expression evaluators.
+///
+/// @ingroup Parser
+///
+/// @par Copyright (c) 2018 vcdMaker team
+///
+/// Permission is hereby granted, free of charge, to any person obtaining a
+/// copy of this software and associated documentation files (the "Software"),
+/// to deal in the Software without restriction, including without limitation
+/// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+/// and/or sell copies of the Software, and to permit persons to whom the
+/// Software is furnished to do so, subject to the following conditions:
+///
+/// The above copyright notice and this permission notice shall be included
+/// in all copies or substantial portions of the Software.
+///
+/// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+/// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+/// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+/// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+/// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+/// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+/// IN THE SOFTWARE.
+
+#include <sstream>
+
+#include "Evaluator.h"
+
+#undef yyFlexLexer
+/// Decimal lexer definition.
+#define yyFlexLexer DecimalFlexLexer
+#include <FlexLexer.h>
+#include "DecimalScanner.h"
+
+#undef yyFlexLexer
+/// Float lexer definition.
+#define yyFlexLexer FloatFlexLexer
+#include <FlexLexer.h>
+#include "FloatScanner.h"
+
+#undef yyFlexLexer
+/// String lexer definition.
+#define yyFlexLexer StringFlexLexer
+#include <FlexLexer.h>
+#include "StringScanner.h"
+
+PARSER::Evaluator::Evaluator(const std::string &rSourceName) :
+    m_rSourceName(rSourceName),
+    m_Context(nullptr, 0)
+{
+}
+
+void PARSER::Evaluator::ParseDecimalString(const std::string &input)
+{
+    std::istringstream iss(input);
+    if (!ParseDecimalStream(iss))
+    {
+        throw PARSER::EXCEPTIONS::ParsingError(ParsingErrorMessage(input));
+    }
+    m_Expression = input;
+}
+
+void PARSER::Evaluator::ParseFloatString(const std::string &input)
+{
+    std::istringstream iss(input);
+    if (!ParseFloatStream(iss))
+    {
+        throw PARSER::EXCEPTIONS::ParsingError(ParsingErrorMessage(input));
+    }
+    m_Expression = input;
+}
+
+void PARSER::Evaluator::ParseStringString(const std::string &input)
+{
+    std::istringstream iss(input);
+    if (!ParseStringStream(iss))
+    {
+        throw PARSER::EXCEPTIONS::ParsingError(ParsingErrorMessage(input));
+    }
+    m_Expression = input;
+}
+
+void PARSER::Evaluator::SetContext(const std::smatch *pGroups, uint64_t lineNo) const
```

Comment:
Suddenly, one-liners are in cpp file but super long functions are in headers...

Reply:
"Set..." function marked `const`. Bad.

Reply:
Fixed.

Reply:
I've noticed that this fix removed like half of the `const` methods in the code. :-D What happened there? It's a bit hard to understand from the code why removal of this one `const` broke const-correctness in so many places.

Reply:
Absolutely right. I got too far there. Although cleaning it up helped me to understand where I needed const' and the way I could get rid of 'mutable'.

Reply:
Great!